lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -159,7 +159,11 @@ export const migrateGranuleRecord = async (
     throw new InvalidArgument('Invalid migration parameters detected, migrateOnlyFiles cannot be set to true if migrateAndOverwrite is also set to true');
   }
 
-  const granule = await translateApiGranuleToPostgresGranule(record, trx);
+  const granule = await translateApiGranuleToPostgresGranule({
+    dynamoRecord: record,
+    knexOrTransaction: trx,
+    preserveNilProperties: true,
+  });
```

Comment:
The call signature changed. It's now an object because we have a million arguments. That's why the number of files in this PR blew up.

Reply:
This makes sense, although I might argue for a named method that flips an internal flag to avoid truthyness/mislabeled keys *outside* of typescripted parameters.     It would be really easy to do something like 


```
  const granule = await translateApiGranuleToPostgresGranule({
    dynamoRecord: record,
    knexOrTransaction: trx,
    preserveNilBroperties: true,
  })
```

from a .js API call that hasn't been converted yet and wind up with unexpected edge cases.  We should catch that in test, but I think we either need to build the logic flag into different exposed methods *or* do a non-truthy check in `translateApiGranuleToPostgresGranule`.

Reply:
This *doubly*  so because I'm concerned we have top-down testing that is not explicitly covering the divergent logic cases.        We need to be really careful that the upstream tests wouldn't break.

Reply:
I believe 9c8a70e1e4bedc19e7da5dd2467c079030590630 addresses this.